core/base/src/constants/chains.ts

```diff
@@ -110,7 +110,8 @@ export const toChain = (chain: number | string | bigint): Chain => {
       if (isChainId(chain)) return chainIdToChain(chain);
       break;
     case "bigint":
-      if (isChainId(Number(chain))) return chainIdToChain(Number(chain) as ChainId);
+      if (chain < (1n << 16n) && isChainId(Number(chain)))
+        return chainIdToChain.get(Number(chain))!;
```

Comment:
yt we should add a `encoding.bignum.downcast` or something that checks for safety first?

Reply:
Yeah, I think that's a great idea too!